--- git status ---
On branch balance
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   isaaclab.sh
	modified:   scripts/environments/list_envs.py
	modified:   source/isaaclab/isaaclab/managers/reward_manager.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/__init__.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/balance/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/isaaclab.sh b/isaaclab.sh
index aa577083..bf5935c7 100755
--- a/isaaclab.sh
+++ b/isaaclab.sh
@@ -17,6 +17,9 @@ tabs 4
 
 # get source directory
 export ISAACLAB_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+export PYTHONPATH="${ISAACLAB_PATH}/source"
+
+
 
 #==
 # Helper functions
diff --git a/scripts/environments/list_envs.py b/scripts/environments/list_envs.py
index 43393d52..4a3b9f26 100644
--- a/scripts/environments/list_envs.py
+++ b/scripts/environments/list_envs.py
@@ -28,6 +28,8 @@ import gymnasium as gym
 from prettytable import PrettyTable
 
 import isaaclab_tasks  # noqa: F401
+# from isaaclab_tasks.manager_based.locomotion import balance  # this is sufficient
+
 
 
 def main():
diff --git a/source/isaaclab/isaaclab/managers/reward_manager.py b/source/isaaclab/isaaclab/managers/reward_manager.py
index 7255d483..2fb45d62 100644
--- a/source/isaaclab/isaaclab/managers/reward_manager.py
+++ b/source/isaaclab/isaaclab/managers/reward_manager.py
@@ -91,6 +91,7 @@ class RewardManager(ManagerBase):
     @property
     def active_terms(self) -> list[str]:
         """Name of active reward terms."""
+        
         return self._term_names
 
     """
@@ -154,6 +155,19 @@ class RewardManager(ManagerBase):
             # Update current reward for this step.
             self._step_reward[:, self._term_names.index(name)] = value / dt
 
+        # for name, term_cfg in zip(self._term_names, self._term_cfgs):
+        #     if term_cfg.weight == 0.0:
+        #         continue
+        #     value = term_cfg.func(self._env, **term_cfg.params) * term_cfg.weight * dt
+            
+        #     # ? Add this line:
+        #     print(f"[{name}] returned shape: {value.shape}")
+            
+        #     self._reward_buf += value  # ? crash will happen here if shape is wrong
+        #     self._episode_sums[name] += value
+        #     self._step_reward[:, self._term_names.index(name)] = value / dt
+
+
         return self._reward_buf
 
     """
diff --git a/source/isaaclab_tasks/isaaclab_tasks/__init__.py b/source/isaaclab_tasks/isaaclab_tasks/__init__.py
index 7de87f53..e8f6add4 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/__init__.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/__init__.py
@@ -23,6 +23,8 @@ __version__ = ISAACLAB_TASKS_METADATA["package"]["version"]
 ##
 
 from .utils import import_packages
+from isaaclab_tasks.manager_based.locomotion import balance  # ? manually import
+
 
 # The blacklist is used to prevent importing configs from sub-packages
 _BLACKLIST_PKGS = ["utils", ".mdp"]